### ISSUE TITLE
Add option to disable search crawling on shared documents

### ISSUE THREAD

**Issue by tommoor** [outline/outline#7486]
It would be nice to have a toggle to add a robots metatag that will discourage search indexing on shared documents.

We can default to indexing being enabled as this is already the case.

**Comment by minaalkhazraji** [comment]
Hi, I think I have a solution. I would like to take this issue, please.

**Comment by infinite-persistence** [comment]
1. Single workspace toggle to disable for all shared docs (`/s/`), or
2. Per-page toggle for each shared doc (multiple `Disallow`s)?

If the latter, I assume there will be a way to query all shared documents in a workspace (is it somewhere in `server/model/Share`?), and would be fine to do here performance-wise?

**Comment by tommoor** [comment]
I'm not really following why querying all shared documents would be needed here. It boils down to:

- A new column on `share` to represent whether search indexing is enabled or not (default: true)
- New UI in the share dialog to allow toggling this setting, UI changes are always the tricky bit to get right. For now it could be an additional toggle that is shown when "Web" is enabled
- Injection of the `no-index` header into shared pages when the preference is set. All of the logic around that is [here](https://github.com/outline/outline/blob/27cbf5a56a35d4f8db3574dd8fab12cda1c7ca61/server/routes/app.ts#L178)

**Comment by infinite-persistence** [comment]
Ah, the meta approach. I was thinking on the robots side :sweat: , something like...
```
User-agent: *
Disallow: /s/specific-page-1
Disallow: /s/specific-page-2
...
```
recalling a bit, meta is indeed the better choice:

> _While Google won't crawl or index the content blocked by robots.txt, we might still find and index a disallowed URL from other places on the web. As a result, the URL address and, potentially, other publicly available information such as anchor text in links to the site can still appear in Google search results. You can stop your URL from appearing in Google Search results completely by using other URL blocking methods, such as password-protecting the files on your server or **using the noindex meta tag or response header.**_
